src/components/CountryPage/CountryPage.jsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useEffect, useContext } from "react";
+import { useParams } from "react-router-dom";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowLeft } from "@fortawesome/free-solid-svg-icons";
+import {
+  CountryPageWrapper,
+  StyledLink,
+  CountryInformationContainer,
+  FlagContainer,
+  NameAndDetails,
+  Details,
+} from "./CountryPage.styles";
+import addCommas from "../../utils/addCommas";
+import { DarkModeContext } from "../../Contexts/DarkModeContext";
+import BorderCountries from "../BorderCountries";
+
+export default function CountryPage() {
+  const { isDarkMode } = useContext(DarkModeContext);
+  const [country, setCountry] = useState();
+  const params = useParams();
+  const checkLocalStorage = () => {
+    const localStorageItem = localStorage.getItem(`${params.countryCode}`);
```

Comment:
w ramach treningu praca z localstorage jest super, ale generalnie takie rozwiązanie nie przeszłoby w komercyjnym projekcie z tego względu, że API i to co zwracają zmienia się tak często, że zapisanie jakiegoś stanu mogłoby następnego dnia zwracać nieprawidłowe dane. W dodatku gdy w kodzie na froncie się coś zmieni to pobranie danych z local storage mogłoby nie pasować do aktualnej wersji frontu i po prostu apka się wysypie

Reply:
Komercyjnie działa to tak, że każde wejście na stronę danego kraju to byłob fetch i ustawienie od nowa danych tego kraju

Reply:
darkMode może siedzieć w local storage - jest to niekrytyczne dla działania aplikacji i też szansa na wysypanie się apki praktycznie jest zerowe